leetcode/DIRECTORY.md

```diff
@@ -0,0 +1,9 @@
+
```

Comment:
The directory file is generated by a script/[GitHub workflow](https://github.com/TheAlgorithms/C/blob/master/.github/workflows/leetcode_directory_writer.yml) automatically.
Would be great if you could add that in this PR so that it doesn't have to be built manually. 🙂